app/models/switch.rb

```diff
@@ -8,6 +8,11 @@ class Switch < ApplicationRecord
   has_many :lans, :dependent => :destroy
   has_many :subnets, :through => :lans
 
+  has_one :hardware, :dependent => :destroy
+
+  belongs_to :ext_management_system, :foreign_key => :ems_id, :inverse_of => :switches,
+    :class_name => "ManageIQ::Providers::PhysicalInfraManager"
```

Comment:
I think this `:class_name` is too specific for the base `Switch` model.

Reply:
Hi @bdunne. Sorry for the delay. I am moving that to the provider and changing here to point to "ExtManagementSystem".

Reply:
@bdunne, I've updated the code. What do you think about?